### PR TITLE
chore: fix stuck runUploadJobAllocator

### PR DESCRIPTION
**Pull request by cisse21**
# Description

Fix stuck warehouse module

## Linear Ticket
Fixes PIPE-1627

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

**Review by achettyiitr — APPROVED** [review]

File: warehouse/router/router.go [hunk 1]
```diff
@@ -377,18 +376,9 @@ loop:
...
-				errors.Is(err, context.DeadlineExceeded),
-				errors.As(err, &pqErr) && pqErr.Code == "57014":
-				break loop
-			default:
-				r.logger.Errorf(`Error executing uploadsToProcess: %v`, err)
-				return err
-			}
+		if err != nil && ctx.Err() == nil {
+			r.logger.Errorf("Error getting uploads to process: %v", err)
+			break loop
```
Comment (Sidddddarth) [inline warehouse/router/router.go:381]: shouldn't we return `error` if context isn't cancelled?

Comment (ktgowtham) [reply warehouse/router/router.go:381]: exactly. isn't the break loop that caused issue in free tier?

Comment (cisse21) [reply warehouse/router/router.go]: fixed it. my bad

**Review by ktgowtham — APPROVED** [review]

**Review by Sidddddarth — APPROVED** [review]

**Review by achettyiitr — APPROVED** [review]

File: warehouse/router/router.go [hunk 2]
```diff
@@ -377,18 +376,9 @@ loop:
...
-			case errors.Is(err, context.Canceled),
-				errors.Is(err, context.DeadlineExceeded),
-				errors.As(err, &pqErr) && pqErr.Code == "57014":
-				break loop
-			default:
-				r.logger.Errorf(`Error executing uploadsToProcess: %v`, err)
-				return err
-			}
+		if err != nil && ctx.Err() == nil {
+			r.logger.Errorf("Error getting uploads to process: %v", err)
```
Comment (lvrach) [inline warehouse/router/router.go:380]: [minor] We should convert this to Errorn, make it easier to filter and setup alerts on

**Pull request merged**